app/crawler/ytn/api.py

```diff
@@ -0,0 +1,111 @@
+# app/crawler/ytn/ytn_api.py
+
+import asyncio
+import json
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+import aiohttp
+import fake_useragent
+
+from app.crawler.base import Article, BaseNewsCrawler
+from common.utils.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+class YtnNewsApiCrawler(BaseNewsCrawler):
+    """
+    YTN 정치 뉴스 크롤러
+    """
+
+    API_URL = "https://www.ytn.co.kr/ajax/getMoreNews.php"
+    USER_AGENT = fake_useragent.UserAgent()
+
+    @staticmethod
+    def random_user_agent() -> str:
+        return YtnNewsApiCrawler.USER_AGENT.random
+
+    async def fetch_articles(self) -> List[Article]:
+        logger.info("[YTN] 정치 뉴스 API 요청 시작")
+
+        timeout = aiohttp.ClientTimeout(total=30)
+        connector = aiohttp.TCPConnector(limit=10, ssl=False)
+
+        articles: List[Article] = []
+
+        # 오늘 날짜 가져오기
+        today = datetime.now().strftime("%Y-%m-%d")
+        logger.info(f"[YTN] 오늘 날짜: {today}")
+
+        async with aiohttp.ClientSession(
+            connector=connector, timeout=timeout
+        ) as session:
+            tasks = []
+            for i in range(1, 4):  # 1~3 페이지 요청
+                tasks.append(self._fetch_news(session, i))
+
+            results = await asyncio.gather(*tasks)
+
+            for result in results:
+                if result and "data" in result:
+                    page_data = result["data"]
+                    for item in page_data.get("data", []):
+                        title = item.get("title")
+                        join_key = item.get("join_key")
+                        # 날짜 정보 추출 ('3' 키에 날짜가 저장되어 있음)
+                        date_str = item.get("3", "")
+
+                        # 오늘 날짜의 기사만 추가
+                        if date_str == today:
+                            if title and join_key:
+                                link = f"https://www.ytn.co.kr/_ln/0101_{join_key}"
+                                articles.append(
+                                    {"title": title, "link": link, "date": date_str}
+                                )
```

Comment:
Article이 dict를 상속받은 클래스기 때문에 이렇게 사용해도 문제는 없어보입니다.
오히려 유연한점이 장점이긴한데 타입체크를 도입하면 예외로 허용해야하는 문제가 생깁니다.

이런 이유로 단순히 딕셔너리 타입으로 추가하는건 경우에따라 좋지 않을 수 있습니다.

제안할 수 있는 다른 대안들

- dataclasses.dataclass
- typing.TypedDict
- pydantic.BaseClass

```suggestion
article = Article(
    title=title,
    link=link,
)
articles.append(article)
```

작성해주신 Article을 사용하려면 이 방식을 제안드려요

Reply:
mbc의 articles 도 같은 방식이었어요